Sources/NIO/SocketChannel.swift

```diff
@@ -1106,10 +1106,18 @@ final class ServerSocketChannel: BaseSocketChannel<ServerSocket> {
         assert(eventLoop.inEventLoop)
 
         let ch = data.forceAsOther() as SocketChannel
-        ch.register().map {
-            ch.becomeActive0()
-            ch.readIfNeeded0()
-        }.whenFailure { error in
+        if isOpen {
```

Comment:
I think it would be nicer to `guard self.isOpen else { ...close }` to match the usual style